src/builder/specification.md

```diff
@@ -0,0 +1,217 @@
+    payload: ExecutionPayloadHeader
+    value: uint256
+```
+
+##### `builder_getPayloadV1` Request
+
+###### `SignedBlindBeaconBlock`
+
+```python
+class SignedBlindBeaconBlock(Container):
+    message: BlindBeaconBlock
+    signature: BLSSignature
+```
+
+###### `BlindBeaconBlock`
+
+```python
+class BlindBeaconBlock(Container):
+    slot: Slot
+    proposer_index: ValidatorIndex
+    parent_root: Root
+    state_root: Root
+    body: BlindBeaconBlockBody
+```
+
+###### `BlindBeaconBlockBody`
+
+```python
+class BlindBeaconBlockBody(Container):
+    randao_reveal: BLSSignature
+    eth1_data: Eth1Data
+    graffiti: Bytes32
+    proposer_slashings: List[ProposerSlashing, MAX_PROPOSER_SLASHINGS]
+    attester_slashings: List[AttesterSlashing, MAX_ATTESTER_SLASHINGS]
+    attestations: List[Attestation, MAX_ATTESTATIONS]
+    deposits: List[Deposit, MAX_DEPOSITS]
+    voluntary_exits: List[SignedVoluntaryExit, MAX_VOLUNTARY_EXITS]
+    sync_aggregate: SyncAggregate
+    execution_payload: ExecutionPayloadHeader
+```
+
+## Errors
+
+The list of error codes introduced by this specification can be found below.
+
+| Code | Message | Meaning |
+| - | - | - |
+| -32000 | Server error | Generic client error while processing request. |
+| -32001 | Unknown hash | No block with the provided hash is known. |
+| -32002 | Unknown validator | No known mapping between validator and feeRecipient. |
+| -32003 | Invalid SSZ | Unable to decode SSZ. |
+| -32004 | Unknown block | Block does not match the provided header. |
+| -32005 | Invalid signature | Provided signature is invalid. |
+| -32006 | Invalid timestamp | Provided timestamp was invalid. |
+| -32600 | Invalid request | The JSON sent is not a valid Request object. |
+| -32601 | Method not found | The method does not exist / is not available. |
+| -32602 | Invalid params | Invalid method parameter(s). |
+| -32603 | Internal error | Internal JSON-RPC error. |
+| -32700 | Parse error | Invalid JSON was received by the server. |
+
+## Routines
+
+### Signing
+
+All signature operations should follow the [standard BLS operations][bls] interface defined in `consensus-specs`.
+
+There are two types of data to sign over in the Builder API:
+* In-protocol messages, e.g. [`BlindBeaconBlock`](#blindbeaconblock), which should compute the signing root using [`compute_signing_root`][compute-signing-root] and use the domain specified for beacon block proposals.
+* Builder API messages, e.g. [`builder_setFeeRecipientV1`](#builder_setFeeRecipientV1) and the response to [`builder_getHeader`](#response-2), which should compute the signing root using [`compute_signing_root`][compute-signing-root] and the domain `DomainType('0xXXXXXXXX')` (TODO: get a proper domain).
+
+As `compute_signing_root` takes `SSZObject` as input, client software should convert in-protocol messages to their SSZ representation to compute the signing root and Builder API messages to the SSZ representations defined [above](#sszobjects).
+
+## Methods
```

Comment:
Missing health endpoint point?

Reply:
Yes! See also https://github.com/flashbots/mev-boost/issues/60 -- the only question is about naming, and which response. Did we leave the workshop with the consensus that a JSON-RPC call is fine?

If so, we could call the method `mevboost_health` or `mevboost_status` perhaps, and the response be just `true` for now? I feel like reusing the `builder` namespace isn't great, because it's literally querying the mevgeth status, not the builder status 🤔

Reply:
We did agree that a JSON-RPC call was fine, but did not come up with a name. It does seem like `builder` isn't a perfect fit, but I don't think it's awkward enough to need a new namespace. `builder_status` seems like an arbitrary name that can be used by both `mev-boost` and relays, should they choose, to provide general status information.